From 2fa1640ea97d3080104b323da46491b71808c25b Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Wed, 22 Mar 2017 10:31:04 +0800
Subject: [PATCH 1605/5242] MLK-14498-10 dma: imx-sdma: fix wrong
 chn_real_count in cyclic mode

commit  cd77d0162c09cadd0629d587226ec317c4980e70 from
https://source.codeaurora.org/external/imx/linux-imx.git

The sdmac->chn_real_count is equal to sdmac->period_len in dma cyclic
mode that is not correct, correct it to real count in current BD transfer.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |    6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index bb3d926..f4a5208 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -791,12 +791,6 @@ static void sdma_update_channel_loop(struct sdma_channel *sdmac)
 		desc->buf_ptail = desc->buf_tail;
 		desc->buf_tail = (desc->buf_tail + 1) % desc->num_bd;
 
-		if (sdmac->peripheral_type == IMX_DMATYPE_UART) {
-			/* restore mode.count after counter readed */
-			sdmac->chn_real_count = bd->mode.count;
-			bd->mode.count = sdmac->chn_count;
-		}
-
 		if (error)
 			sdmac->status = old_status;
 		/*
-- 
1.7.9.5

